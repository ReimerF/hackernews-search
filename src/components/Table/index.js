import React from "react";
import Button from "../Button";
import PropTypes from "prop-types";

const largeColumn = {
  width: "40%"
};
const midColumn = {
  width: "30%"
};
const smallColumn = {
  width: "10%"
};

const Table = ({ list, onDismiss, onSort, sortKey, SORTS }) => (
  <div className="table">
    <div key="header-row" className="table-row">
      <span style={largeColumn} onClick={() => onSort("TITLE")}>
        Title
      </span>
      <span style={midColumn} onClick={() => onSort("AUTHOR")}>
        Author
      </span>
      <span style={smallColumn} onClick={() => onSort("COMMENTS")}>
        Comments
      </span>
      <span style={smallColumn} onClick={() => onSort("POINTS")}>
        Points
      </span>
      <span style={smallColumn} onClick={() => onSort("NONE")}>
        Options
      </span>
    </div>
    {SORTS[sortKey](list).map(item => (
      <div key={item.objectID} className="table-row">
        <span style={largeColumn}>
          <a href={item.url}>{item.title}</a>
        </span>
        <span style={midColumn}>{item.author}</span>
        <span style={smallColumn}>{item.num_comments}</span>
        <span style={smallColumn}>{item.points}</span>
        <Button
          onClick={() => onDismiss(item.objectID)}
          className={"button-inline"}
        >
          {" "}
          Dismiss{" "}
        </Button>
      </div>
    ))}
  </div>
);

Table.propTypes = {
  list: PropTypes.array.isRequired,
  onDismiss: PropTypes.func.isRequired
};

export default Table;
