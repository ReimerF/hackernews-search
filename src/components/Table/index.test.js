import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Table from "./index";

describe("Table", () => {
  const props = {
    list: [
      {
        title: "1",
        author: "2",
        num_comments: "3",
        points: "4",
        objectID: "x"
      },
      { title: "5", author: "6", num_comments: "7", points: "8", objectID: "y" }
    ],
    onDismiss: jest.fn()
  };

  it("shows two items in list", () => {
    const element = shallow(<Table {...props} />);
    expect(element.find(".table-row").length).toBe(2);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<Table {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
