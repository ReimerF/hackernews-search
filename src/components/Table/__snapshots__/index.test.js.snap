// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table has a valid snapshot 1`] = `
<div
  className="table"
>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a>
        1
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      3
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      4
    </span>
    <button
      className="button-inline"
      onClick={[Function]}
      type="button"
    >
       
      Dismiss
       
    </button>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a>
        5
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      6
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      7
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      8
    </span>
    <button
      className="button-inline"
      onClick={[Function]}
      type="button"
    >
       
      Dismiss
       
    </button>
  </div>
</div>
`;
